plugins/org.jkiss.dbeaver.ext.altibase/src/org/jkiss/dbeaver/ext/altibase/model/AltibaseProperty.java

```diff
@@ -0,0 +1,115 @@
+/*
+ * DBeaver - Universal Database Manager
+ * Copyright (C) 2010-2024 DBeaver Corp and others
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.jkiss.dbeaver.ext.altibase.model;
+
+import org.jkiss.code.NotNull;
+import org.jkiss.code.Nullable;
+import org.jkiss.dbeaver.Log;
+import org.jkiss.dbeaver.ext.altibase.AltibaseConstants;
+import org.jkiss.dbeaver.model.exec.jdbc.JDBCResultSet;
+import org.jkiss.dbeaver.model.impl.jdbc.JDBCUtils;
+import org.jkiss.dbeaver.model.meta.Property;
+import org.jkiss.dbeaver.model.struct.DBSObject;
+import org.jkiss.utils.CommonUtils;
+
+public class AltibaseProperty implements DBSObject {
+
+    private static final Log log = Log.getLog(AltibaseProperty.class);
+
+    private final AltibaseDataSource dataSource;
+    private final String name;
+    private String description;
+    private boolean dynamic;
+    private long attr;
+
+    public AltibaseProperty(AltibaseDataSource dataSource, JDBCResultSet dbResult) {
+        final int valueCount;
+        final String min;
+        final String max;
+        String tmp;
+
+        this.dataSource = dataSource;
+        name = JDBCUtils.safeGetString(dbResult, "NAME");
+        valueCount  = JDBCUtils.safeGetInt(dbResult, "STOREDCOUNT");
+        attr = JDBCUtils.safeGetLong(dbResult, "ATTR");
+        min  = JDBCUtils.safeGetString(dbResult, "MIN");
+        max  = JDBCUtils.safeGetString(dbResult, "MAX");
+
+        // configurable without restarting the server
+        dynamic = (((attr & AltibaseConstants.IDP_ATTR_RD_MASK) | AltibaseConstants.IDP_ATTR_RD_WRITABLE) 
+                == AltibaseConstants.IDP_ATTR_RD_WRITABLE);
```

Comment:
I'm not sure if this makes sense. This will always `true`, as you're OR-ing values instead of AND-ing them.

And since you have a mask, you can do this instead:
```suggestion
        dynamic = (attr & AltibaseConstants.IDP_ATTR_RD_MASK) == AltibaseConstants.IDP_ATTR_RD_WRITABLE;
```

Reply:
Thanks for your concern, but it works fine.

https://github.com/ALTIBASE/Documents/blob/master/Manuals/Altibase_7.3/eng/General%20Reference-1.Data%20Types%20%26%20Altibase%20Properties.md

- IDLE_TIMEOUT (Unit: second) Read-Write The value of this property can be changed using the ALTER SYSTEM or ALTER SESSION statement while Altibase is running.

- EXTPROC_AGENT_IDLE_TIMEOUT Attributes Read-Only

![image](https://github.com/dbeaver/dbeaver/assets/32251385/1a67d015-4858-43cf-871b-7532fe414a4b)

Reply:
I understand how it works. But `x | AltibaseConstants.IDP_ATTR_RD_WRITABLE` doesn't make any sense since `IDP_ATTR_RD_WRITABLE` is `0`, so it can be inlined to `x | 0`, which can be further simplified to `x`.